File: app/src/main/java/com/thehappybit/instagramclone/PostListAdapter.java
package com.thehappybit.instagramclone;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.thehappybit.instagramclone.Models.Post;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

public class PostListAdapter  extends RecyclerView.Adapter<PostViewHolder> {

    private final LayoutInflater inflater;

    private List<Post> posts;

    public PostListAdapter(Context context,List<Post> posts) {
        inflater = LayoutInflater.from(context);
        this.posts = posts;
    }

    @NonNull
    @Override
    public PostViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = inflater.inflate(R.layout.feed_item, parent, false);
        return new PostViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull PostViewHolder holder, int position) {

        if (posts != null){
            Post current = posts.get(position);
            holder.bindToPost(current);
        }
    }

    @Override
    public int getItemCount() {
        if (posts != null){
            return posts.size();
        } else
            return 0;
    }

    public void setPosts(List<Post> posts) {
        this.posts = posts;
        notifyDataSetChanged();
    }
}

File: app/src/main/java/com/thehappybit/instagramclone/PostListAdapter.kt
package com.thehappybit.instagramclone

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.thehappybit.instagramclone.models.Post


class PostListAdapter(val context : Context, var posts: List<Post?>) : RecyclerView.Adapter<PostViewHolder>() {

    private val  inflater= LayoutInflater.from(context)





    override fun  onCreateViewHolder(parent:ViewGroup,  viewType:Int):PostViewHolder {
        val view = inflater.inflate(R.layout.feed_item, parent, false)
        return PostViewHolder(view)
    }


    override fun  onBindViewHolder(  holder:PostViewHolder,  position:Int) {

        if (posts != null){
            val  current = posts[position]
            holder.bindToPost(current)
        }
    }

    override fun  getItemCount():Int {
        return if (posts != null){
            posts.size
        } else
            0;
    }

     fun setposts(posts:List<Post?>) {
        this.posts = posts
        notifyDataSetChanged();
    }
}
